src/v/datalake/translation/deps.h

```diff
@@ -21,4 +34,151 @@ class noop_mem_tracker : public writer_mem_tracker {
     void release() override;
 };
 
+class writer_reservations_impl : public writer_mem_tracker {
+public:
+    explicit writer_reservations_impl(
+      scheduling::reservations_tracker& scheduling_reservations)
+      : _reservations_tracker(scheduling_reservations) {}
+
+    ss::future<> maybe_reserve_memory(size_t bytes, ss::abort_source&) override;
+    void update_current_memory_usage(size_t) override;
+    void release() override;
+
+private:
+    size_t _available_memory{0};
+    size_t _total_reserved_memory{0};
+    scheduling::reservations_tracker& _reservations_tracker;
+    chunked_vector<ssx::semaphore_units> _reservations;
+};
+
+class coordinator_api {
+public:
+    coordinator_api() = default;
+    coordinator_api(const coordinator_api&) = delete;
+    coordinator_api& operator=(const coordinator_api&) = delete;
+    coordinator_api(coordinator_api&&) = delete;
+    coordinator_api& operator=(coordinator_api&&) = delete;
+
+    virtual ~coordinator_api() = default;
+
+    virtual ss::future<coordinator::add_translated_data_files_reply>
+      add_translated_data_files(coordinator::add_translated_data_files_request)
+      = 0;
+
+    virtual ss::future<coordinator::fetch_latest_translated_offset_reply>
+      fetch_latest_translated_offset(
+        coordinator::fetch_latest_translated_offset_request)
+      = 0;
+
+    static std::unique_ptr<coordinator_api>
+    make_default_coordinator_api(coordinator::frontend&);
+};
+
+class data_source {
```

Comment:
a comment on top of this class would be great

Reply:
yup, comments are WIP

Reply:
done